### PR TITLE
Upgrade msys packages

### PR DESCRIPTION
**Pull request by cocreature**
The old ones pull in dependencies that have disappeared.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: dev-env/windows/manifests/msys2.json [hunk 1]
```diff
@@ -2,21 +2,21 @@
...
     "url": [
-        "https://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20200903.tar.xz",
+        "https://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20210228.tar.xz",
         "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-jq-1.6-4-any.pkg.tar.zst#/jq.msys2",
         "https://repo.msys2.org/msys/x86_64/gnu-netcat-0.7.1-1-x86_64.pkg.tar.xz#/netcat.msys2",
         "https://repo.msys2.org/msys/x86_64/patch-2.7.6-1-x86_64.pkg.tar.xz#/patch.msys2",
-        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.g-1-any.pkg.tar.xz#/openssl.msys2",
-        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-3.8.2-2-any.pkg.tar.xz#/python.msys2",
+        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.j-1-any.pkg.tar.zst#/openssl.msys2",
+        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-3.8.8-2-any.pkg.tar.zst#/python.msys2",
```
Comment (cocreature) [inline dev-env/windows/manifests/msys2.json:10]: Not entirely sure if those are needed but I see no reason to stick to outdated openssl or python versions.

File: ci/build.yml [hunk 2]
```diff
@@ -158,7 +158,7 @@ jobs:
   timeoutInMinutes: 360
   pool:
     name: 'windows-pool'
-    demands: assignment -equals default
+    demands: assignment -equals windows-debug
```
Comment (cocreature) [inline ci/build.yml:161]: will remove before merging

File: dev-env/windows/manifests/msys2.json (hunk 1 above)
Comment (garyverhaegen-da) [reply dev-env/windows/manifests/msys2.json:10]: Any reason to have fragments in those URLs? IIRC they are not supposed to matter to the server (per HTTP spec), and I can't imagine what scoop might want to do with them.

**Review by garyverhaegen-da — APPROVED** [review]
Very much a case of "if CI is happy, :shipit:".

File: dev-env/windows/manifests/msys2.json (hunk 1 above)
Comment (cocreature) [reply dev-env/windows/manifests/msys2.json:10]: scoop uses them as the file name to download to.

Comment (garyverhaegen-da) [reply dev-env/windows/manifests/msys2.json:10]: Thanks. That is a little bit horrifying...

Comment (cocreature) [reply dev-env/windows/manifests/msys2.json:10]: I agree, I tried to delete it at some point which is how I learned this :slightly_smiling_face:

**Pull request merged**